src/lib.rs

```diff
@@ -632,6 +633,14 @@ impl<'a, 'input> Ord for Node<'a, 'input> {
     }
 }
 
+impl<'a, 'input> Hash for Node<'a, 'input> {
```

Comment:
Can we use anonymous lifetimes here?

```
impl Hash for Node<'_, '_> {
```

Reply:
Looks like we can! I can change that. Looks like it can be done for Eq, PartialEq, Ord, PartialOrd impls as well. Does it make sense to anonymize those as well? Perhaps in a separate PR?

Reply:
I guess it will be better to do via a separate request.